sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/Models/NotificationHookModelTests.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.AI.MetricsAdvisor.Administration;
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+
+namespace Azure.AI.MetricsAdvisor.Tests
+{
+    public class NotificationHookModelTests : MockClientTestBase
+    {
+        public NotificationHookModelTests(bool isAsync) : base(isAsync)
+        {
+        }
+
+        private string UnknownHookContent => $@"
+        {{
+            ""hookId"": ""{FakeGuid}"",
+            ""hookName"": ""unknownHookName"",
+            ""hookType"": ""unknownType"",
+            ""externalLink"": ""https://fakeuri.com/"",
+            ""description"": ""unknown hook description"",
+            ""admins"": [
+              ""foo@contoso.com""
+            ]
+        }}
+        ";
+
+        [Test]
+        public async Task NotificationHookGetsUnknownHook()
+        {
+            MockResponse getResponse = new MockResponse(200);
+            getResponse.SetContent(UnknownHookContent);
+
+            MetricsAdvisorAdministrationClient adminClient = CreateInstrumentedAdministrationClient(getResponse);
+            NotificationHook hook = await adminClient.GetHookAsync(FakeGuid);
+
+            Assert.That(hook.Id, Is.EqualTo(FakeGuid));
+            Assert.That(hook.Name, Is.EqualTo("unknownHookName"));
+            Assert.That(hook.ExternalUri.AbsoluteUri, Is.EqualTo("https://fakeuri.com/"));
+            Assert.That(hook.Description, Is.EqualTo("unknown hook description"));
+            Assert.That(hook.AdministratorEmails.Single(), Is.EqualTo("foo@contoso.com"));
+        }
+
+        [Test]
+        public async Task NotificationHookUpdatesUnknownHook()
+        {
+            MockResponse getResponse = new MockResponse(200);
+            getResponse.SetContent(UnknownHookContent);
+
+            MockResponse updateResponse = new MockResponse(200);
+            updateResponse.SetContent(UnknownHookContent);
+
+            MockTransport mockTransport = new MockTransport(getResponse, updateResponse);
+            MetricsAdvisorAdministrationClient adminClient = CreateInstrumentedAdministrationClient(mockTransport);
+            NotificationHook hook = await adminClient.GetHookAsync(FakeGuid);
+
+            hook.Name = "newHookName";
+            hook.ExternalUri = new Uri("https://newfakeuri.com/");
+            hook.Description = "new description";
+
+            await adminClient.UpdateHookAsync(hook);
+
+            MockRequest request = mockTransport.Requests.Last();
+            string content = ReadContent(request);
+
+            Assert.That(request.Uri.Path, Is.EqualTo($"/metricsadvisor/v1.0/hooks/{FakeGuid}"));
```

Comment:
do you really need to validate the uri? I didn't see anything in your code that will modify this. 
Also, consider making the version match the version the user sets in the client so you don't have to update this test every time you update the service version.

Reply:
We're checking the we could get and send back every `NotificationHook` property. We're doing it for the URI as well because that's the only way of validating the `Id` property (we don't send it as part of the JSON body).

You have a point about the version. I'll update the assertion to a `Contains.Substring(FakeGuid)` given that we just want to test if the ID is being sent correctly.